src/subclass/widget.rs

```diff
@@ -209,6 +210,10 @@ pub trait WidgetImpl: WidgetImplExt + ObjectImpl + 'static {
     fn get_preferred_height_for_width(&self, widget: &Widget, width: i32) -> (i32, i32) {
         self.parent_get_preferred_height_for_width(widget, width)
     }
+
+    fn size_allocate(&self, widget: &Widget, allocation: &mut Allocation) {
```

Comment:
Should this be a mutable reference? In GTK it's all const. AFAIU this gives the allocation to the subclass, not returns the allocation from the subclass.

Also the `gtk_widget_size_allocate()` function should probably be exposed at the same time.

Reply:
it is exposed as "gtk::WidgetExt::size_allocate". or am I missing something?

Reply:
> Should this be a mutable reference? In GTK it's all const. AFAIU this gives the allocation to the subclass, not returns the allocation from the subclass.

See PR #961

Reply:
Indeed